src/Admin/UserRESTEndpointExtension.php

```diff
@@ -0,0 +1,122 @@
+<?php
+/**
+ * Class UserRESTEndpointExtension
+ *
+ * @package Ampproject\Ampwp
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AMP_Theme_Support;
+use AmpProject\AmpWP\Infrastructure\Delayed;
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Error;
+use WP_User;
+
+/**
+ * UserRESTEndpointExtension class.
+ *
+ * @since 2.2
+ * @internal
+ */
+class UserRESTEndpointExtension implements Service, Registerable, Delayed {
+
+	/**
+	 * User meta key that stores a template mode for which the "Review" panel was dismissed.
+	 *
+	 * @var string
+	 */
+	const USER_FIELD_REVIEW_PANEL_DISMISSED_FOR_TEMPLATE_MODE = 'amp_review_panel_dismissed_for_template_mode';
+
+	/**
+	 * Get registration action.
+	 *
+	 * @return string
+	 */
+	public static function get_registration_action() {
+		return 'rest_api_init';
+	}
+
+	/**
+	 * Adds hooks.
+	 */
+	public function register() {
+		$this->register_rest_field();
+	}
+
+	/**
+	 * Register REST field for storing a template mode for which the "Review" panel was dismissed.
+	 */
+	public function register_rest_field() {
+		register_rest_field(
+			'user',
+			self::USER_FIELD_REVIEW_PANEL_DISMISSED_FOR_TEMPLATE_MODE,
+			[
+				'get_callback'    => [ $this, 'get_review_panel_dismissed_for_template_mode' ],
+				'update_callback' => [ $this, 'update_review_panel_dismissed_for_template_mode' ],
+				'schema'          => [
+					'description' => __( 'For which template mode the Review panel on the Settings screen was dismissed by a user', 'amp' ),
+					'type'        => 'string',
+					'enum'        => [
+						'',
+						AMP_Theme_Support::READER_MODE_SLUG,
+						AMP_Theme_Support::STANDARD_MODE_SLUG,
+						AMP_Theme_Support::TRANSITIONAL_MODE_SLUG,
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Provides a template mode for which the "Review" panel has been dismissed by a user.
+	 *
+	 * @param array $user Array of user data prepared for REST.
+	 *
+	 * @return string|WP_Error Template mode for which the panel is dismissed, empty string if the option has not been set, or WP_Error if the current user lacks permission.
+	 */
+	public function get_review_panel_dismissed_for_template_mode( $user ) {
+		if ( wp_get_current_user()->ID !== $user['id'] ) {
+			return new WP_Error(
+				'amp_rest_cannot_get_other_user',
+				__( 'Sorry, the current user is not allowed to get this data about other user.', 'amp' ),
+				[ 'status' => rest_authorization_required_code() ]
+			);
```

Comment:
I just realized that this should probably:

```suggestion
			return null;
```

Otherwise, when using the REST API to fetch users, then this field shows up as an error:

> ![image](https://user-images.githubusercontent.com/134745/135313968-e6ce7d5c-e773-4e0b-9e17-dab256f07ab4.png)

I'll need to double-check the best practice for visibility of a field like this.

Reply:
That being said, we currently allow the `amp_dev_tools_enabled` field to be exposed to other users, so perhaps this should be visible by default too?

Reply:
Only `administrator` users are allowed to `list_users` for the `/users` endpoint, and for `/users/:id` a request to non-self also requires `list_users` cap. So it's not a problem to expose this information.

Reply:
Updating this now…